props/nested-components.md

```diff
@@ -0,0 +1,195 @@
+
+Sofia is a teacher creating an attendance app. Her app should display a list of students. At this moment, she starts with this implementation of a `StudentList` component in `src/components/StudentList.js`.
+
+(Class names have been omitted from this code snippet for clarity)
+
+<!-- prettier-ignore-start -->
+```js
+import React from 'react';
+import './StudentList.css';
+
+const StudentList = () => {
+    return (
+        <section>
+            <h2>Student List</h2>
+            <ul>
+                <li>Student A</li>
+                <li>Student B</li>
+                <li>Student C</li>
+            </ul>
+        </section>
+    )
+}
+
+export default StudentList;
+```
+<!-- prettier-ignore-end -->
+
+This component displays a list of students, where each student is in an `<li>` element.
+
+However, Sofia knows that each student list item should display a lot of data, such as their name and email address.
+
+Sofia can apply the container and presentational component design patterns, and:
+
+1. Create a `Student` component, which will act as a presentational component
+1. Use the `Student` component in `StudentList`, which will act as a container component
+
+### `Student` Component
+
+Sofia's `Student` component should display a student's name and email address. She'll create a new file, `src/components/Student.js`, and write this code:
+
+<!-- prettier-ignore-start -->
+```js
+import React from 'react';
+
+const Student = () => {
+    return (
+        <li>
+            <ul>
+                <li>Nickname: Ada</li>
+                <li>Email: ada@dev.org</li>
+            </ul>
+        </li>
+    )
+}
+
+export default Student;
+```
+<!-- prettier-ignore-end -->
+
+### Rendering `Student`s in `StudentList` Component
+
+Now, Sofia can update her `StudentList` component.
+
+First, she needs to import the `Student` component into `src/components/StudentList.js` with `import Student from './Student';`.
+
+Then, she should replace the contents of the `<ul>` element.
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = () => {
+    return (
+        <section>
+            <h2>Student List</h2>
+            <ul>
+                <Student></Student>
+                <Student></Student>
+                <Student></Student>
+            </ul>
+        </section>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+(Imports, exports, and class names have been omitted from this code snippet for clarity)
+
+Sofia's app now looks like this, which includes three rendered `Student` components! (This image also includes a previously implemented `ClassInfo` component.)
+
+![An attendance web app with a heading "Student List," and a list of three students. All three students read "Nickname: Ada" and "Email: ada@dev.org."](../assets/props_nested-components_render.png)  
+_Fig. An attendance web app with a heading "Student List," and a list of three students. All three students read "Nickname: Ada" and "Email: ada@dev.org."_
```

Comment:
revised the alt text and the caption (just minor to the alt)
alt text should describe the image itself (as this does) for folx using a screen reader to understand what the image is of
the caption should provide context for how the image relates to the surrounding text. If it's the same as the alt text, then a screen reader would end up reading the same thing twice. And for sighted folx, having the caption describe the contents of the image is strange, since they can see the image. For both cases, the caption should provide context about why the image is here.

In this case, no fancy caption is required, since it's pretty obvious that we're simply showing the updated output for reference, but we should still avoid describing the picture in words for the caption.

```suggestion
![Sophia's attendance web app with a heading "Student List," and a list of three students. All three students read "Nickname: Ada" and "Email: ada@dev.org."](../assets/props_nested-components_render.png)  
_Fig. Updated `StudentList` output. The previously implemented `ClassInfo` component is also displayed._
```